pkg/provider/provider.go

```diff
@@ -520,3 +534,40 @@ func getObjectVersion(id string) string {
 func getObjectUID(objectName, objectType string) string {
 	return fmt.Sprintf("%s/%s", objectType, objectName)
 }
+
+// validateObjectEncoding checks if the object encoding is valid and is supported
+// for the given object type
+func validateObjectEncoding(objectEncoding, objectType string) error {
+	if len(objectEncoding) == 0 {
+		return nil
+	}
+
+	// ObjectEncoding is supported only for secret types
+	if objectType != VaultObjectTypeSecret {
+		return fmt.Errorf("ObjectEncoding only supported for objectType: secret")
+	}
+
+	if !strings.EqualFold(objectEncoding, objectEncodingHex) && !strings.EqualFold(objectEncoding, objectEncodingBase64) && !strings.EqualFold(objectEncoding, objectEncodingUtf8) {
+		return fmt.Errorf("Invalid objectEncoding: %v, should be hex, base64 or utf-8", objectEncoding)
```

Comment:
nit: Error string should be capitalized

```suggestion
		return fmt.Errorf("invalid objectEncoding: %v, should be hex, base64 or utf-8", objectEncoding)
```